### PR TITLE
Progetto Fiderio

### PR DESCRIPTION
**Pull request by Defenestration123**

### CODE REVIEW

File: 2022/Fiderio/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- Repeat the two previous steps for the "diskInfluxTask.flux" and the "ramInfluxTask.flux" files
+
+# Usage
+
+Execute "SystemMonitorMain.py":
+```sh
+python3.8 SystemMonitorMain.py
+```
+The application is now monitoring, open your browser and navigate to localhost:8086 to check on the data that is being saved to the DB.
+Whenever you're done press CTRL+c to stop the application.
```
Comment (lucaderi) [inline 2022/Fiderio/README.md:50]: Can you also ass
- some screenshot of the expected results
- explain how you have tested the tool
- describe limitations (e.g. Linux only?) and design choices

**Review by lucaderi — CHANGES_REQUESTED** [review]
Please address the requested changes.

File: 2022/Fiderio/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+-Python 3.8 \
+-easysnmp 
+
+You can get InfluxDB from its official site here: https://www.influxdata.com/get-influxdb/ \
+
+For easysnmp you need to open the terminal and type the following command:
+```sh
+sudo apt install snmp snmpd snmp-mibs-downloader libsnmp-dev
+```
+then edit the snmp.conf file (by default found in: /etc/snmp/) so that loading MIBs is allowed.
```
[the remark this answers is not in the document]
Comment (Defenestration123) [reply 2022/Fiderio/README.md]: Readme file updated.

File: 2022/Fiderio/System Monitor/SystemMonitorMain.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        .tag("host", "host1") \
+        .field("CPU Percent Usage", cpuValues[-1]) \
+        .field("Next Value Prediction", int(cpuPrediction[-1])) \
+        .field("Lower Bound", int(cpuLowBound)) \
+        .field("Higher Bound", int(cpuHighBound)) \
+        .time(datetime.utcnow(), WritePrecision.NS)
+        write_api.write(bucket, org, point)
+
+        #endif
+    time.sleep(2)
```
[the remark this answers is not in the document]
Comment (Defenestration123) [reply 2022/Fiderio/System Monitor/SystemMonitorMain.py]: This is just to limit how often the software reads new values, does the smoothing and sends data to influxdb.
A new prompt has been added to let the user choose the frequency.

File: 2022/Fiderio/System Monitor/SystemMonitorMain.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  except KeyboardInterrupt:
+        print('\nClosing...')
+
+        #InfluxDB cleanup
+        start = "1970-01-01T00:00:00Z"
+        stop = datetime.utcnow()
+        delete_api.delete(start, stop, '_measurement="ramMonitor"', bucket=bucket, org=org)
+        delete_api.delete(start, stop, '_measurement="diskMonitor"', bucket=bucket, org=org)
+        delete_api.delete(start, stop, '_measurement="cpuMonitor"', bucket=bucket, org=org)
+        time.sleep(1)
```
[the remark this answers is not in the document]
Comment (Defenestration123) [reply 2022/Fiderio/System Monitor/SystemMonitorMain.py]: A new prompt has been added to let the user decide whether they want to delete the old data or not.

File: 2022/Fiderio/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+- Open "configFile.py" inside the System Monitor folder and insert your InfluxDB token, org and bucket
+- Make sure the Influxd service is running, open your browser and navigate to localhost:8086, then click on tasks in the sidebar and create a new task
+- Copy the script found in "cpuInfluxTask.flux" into your new task and make sure to set proper values for the "bucketName", "tokenTel" and "channelTel" variables
+- Repeat the two previous steps for the "diskInfluxTask.flux" and the "ramInfluxTask.flux" files
+
+# Usage
+
+Execute "SystemMonitorMain.py":
+```sh
+python3.8 SystemMonitorMain.py
```
[the remark this answers is not in the document]
Comment (Defenestration123) [reply 2022/Fiderio/README.md:55]: Readme file updated.

File: 2022/Fiderio/System Monitor/SystemMonitorMain.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+#Connecting to InfluxDB
+client=InfluxDBClient(url="http://localhost:8086", token=token, org=org)
+write_api=client.write_api(write_options = SYNCHRONOUS)
+delete_api = client.delete_api()
+
+#Values for double exponential smoothing and deviation calculations
+alpha = 0.5
+beta = 0.5
+rho = 0.5
+
```
[the remark this answers is not in the document]
Comment (Defenestration123) [reply 2022/Fiderio/System Monitor/SystemMonitorMain.py:32]: An alpha value of 0.5 means that , in the calculation of the "Level", we give equal weight to both new values read and the previous prediction ( previous level + previous trend).
A beta value of 0.5 means that, in the calculation of the "Trend", we give equal weight to the difference between the current level value and the previous level value and to the older trend value.
The value we give to rho determines how far away both the "High bound" and the "Low bound" value are from the actual value read.

This all means that in determining the prediction for the next value we are neither too inflexible nor too flexible and that was the entire thought process behind choosing "0.5".

File: 2022/Fiderio/README.md (hunk 1 above)
Comment (Defenestration123) [reply 2022/Fiderio/README.md:60]: Readme updated.

**Pull request merged**